command/agent/template/template.go

```diff
@@ -264,32 +264,15 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		ServerName: pointerutil.StringPtr(""),
 	}
 
-	// The cache does its own retry management based on sc.AgentConfig.Retry,
-	// so we only want to set this up for templating if we're not routing
-	// templating through the cache.  We do need to assign something to Retry
-	// though or it will use its default of 12 retries.
+	// We need to assign something to Vault.Retry or it will use its default of 12 retries.
+	// This retry value will be respected regardless of if we use the cache.
 	var attempts int
 	if sc.AgentConfig.Vault != nil && sc.AgentConfig.Vault.Retry != nil {
 		attempts = sc.AgentConfig.Vault.Retry.NumRetries
 	}
 
 	// Use the cache if available or fallback to the Vault server values.
 	if sc.AgentConfig.Cache != nil {
-		attempts = 0
-
```

Comment:
my only comment is that is there any test for this part to make sure the functionality satisfies the requirement?

Reply:
All of the current tests pass, and there are at least a lot of caching tests. I do regret not having a test for this PR but it doesn't look particularly easy to do one - the testing framework doesn't seem particularly set up for testing retries at all (it would require framework changes or something larger). I feel like that's something we should tackle in the future but I didn't feel like that should block this change. Open to contrasting opinions, though.

Reply:
That is fair. 
And, it seems based on the PR description, you have manually verified the change as well. I agree, adding a go test should not block this PR!